ring_doorbell/__init__.py

```diff
@@ -104,22 +105,28 @@ def _authenticate(self, attempts=RETRY_TOKEN, session=None):
         """Authenticate user against Ring API."""
         url = API_URI + NEW_SESSION_ENDPOINT
         loop = 0
+        # make a copy as we're mutating headers in the loop below
+        # which would cause issues with _get_oauth_token()
+        # which expects a non mutated HEADERS copy
+        modified_headers = HEADERS.copy()
         while loop <= attempts:
-            HEADERS['Authorization'] = \
+            modified_headers['Authorization'] = \
                 'Bearer {}'.format(self._get_oauth_token())
             loop += 1
+
             try:
                 if session is None:
                     req = self.session.post((url),
                                             data=POST_DATA,
-                                            headers=HEADERS)
+                                            headers=modified_headers)
                 else:
                     req = session
             except requests.exceptions.RequestException as err_msg:
                 _LOGGER.error("Error!! %s", err_msg)
                 raise
 
             if not req:
+                time.sleep(5)  # add a pause or you'll get rate limited (429)
```

Comment:
@tchellomello any thoughts on if I should just remove this line? eg. let the rate limiting occur and/or keep looping as a failure here is unlikely to clear up after a retry?

Reply:
@xernaj I see your point. Let's keep it (even though we are blocking I/O for now) but let's transform this as a variable to be passed to the method, so we can users can increase or remove it as necessary.  Can you make this change too?

Reply:
I've updated it with a var "wait" which is default to 1.0.